File: backend/schemas/risk.py
from pydantic import BaseModel


class RiskScore(BaseModel):
    score: int
    grade: str
    breakdown: dict
    finding_count_by_severity: dict[str, int]


class GraphNode(BaseModel):
    id: str
    type: str
    label: str
    severity: str | None


class GraphEdge(BaseModel):
    source: str
    target: str
    relation: str


class RiskGraph(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]

File: frontend/src/types/risk.ts
export interface RiskScore {
  score: number
  grade: string
  breakdown: {
    findings: Record<string, number>
    cves: Record<string, number>
    raw_score: number
  }
  finding_count_by_severity: Record<string, number>
}

export interface GraphNode {
  id: string
  type: 'analysis' | 'finding' | 'library' | 'cve' | 'host' | 'component' | 'permission'
  label: string
  severity: string | null
}

export interface GraphEdge {
  source: string
  target: string
  relation: string
}

export interface RiskGraph {
  nodes: GraphNode[]
  edges: GraphEdge[]
}
